TipulDay.py

```diff
@@ -1,178 +1,182 @@
-            print("Error 5: Issue requesting from the website.")
-            FailedForms=FailedForms+1
+            print("[!] Error 5: Issue requesting from the website")
+            FailedForms += 1
             continue
 
-        #parsing the response using BeautifulSoup
+        # parsing the response using BeautifulSoup
         soup = BeautifulSoup(response.text, 'html.parser')
 
-        #finding the form elements in the page 
+        # finding the form elements in the page
         form_elements = soup.find_all('input')
         #form_elements.append(soup.find_all('input', {'type': 'tel'}))
-        #creating a dictionary to store the form data
+        # creating a dictionary to store the form data
         form_data = {}
 
-        #looping through the form elements and getting the data 
+        # looping through the form elements and getting the data
         for element in form_elements:
-                name = element.get('name')
-                value = element.get('value')
-                form_data[name] = value
+            name = element.get('name')
+            value = element.get('value')  # why is the value necessary?
+            form_data[name] = value
 
-    
         for data in form_data:
-                    DataName =data
-                    try:
-                        if "name" in DataName:
-                                form_data[DataName] = FormNameValue
-                        if "phone" in DataName or "num" in DataName or "tel" in DataName:
-                                form_data[DataName] = FormPhoneNumberValue  
-                        if "mail" in DataName:
-                                form_data[DataName] = FormEmailValue   
-                        if "Comment"in DataName or "message"in DataName: 
-                                form_data[DataName] = FormCommentValue  
-                    except:
-                        print('Error 6: Some parameters might be missing:',website,'.')
-                         
-
-
-                #making a post request to the website with the form data 
+            DataName = data
+            try:
+                if "name" in DataName:  # Some developers like to name the fields fname or lname so this will miss
+                    form_data[DataName] = FormNameValue
+                # You are searching with a constant string field-names that can be Phone or phoneNumber and this line wont recognize them
+                if "phone" in DataName or "num" in DataName or "tel" in DataName:
+                    form_data[DataName] = FormPhoneNumberValue
+                if "mail" in DataName:
+                    form_data[DataName] = FormEmailValue
+                if "Comment" in DataName or "message" in DataName:
+                    form_data[DataName] = FormCommentValue
+                # To make it simpler try to create a regex to match the names of these inputs and not strings. In addition the 'type' attribute can help you find this information faster
```

Comment:
any code ideas?

Reply:
I will create a seperate branch with the new code